### DISCUSSION TITLE
Does YAMCS plugin support XTCE array type?

### DISCUSSION THREAD

**Discussion by mk51tun** [nasa/openmct#7723] [Community Q&A]
Hello all,

We are new to YAMCS server and YAMCS plugin. 

We are using an XTCE array type in our YAMCS server.  We created an array of aggregates.  The aggregate contains four parameters, latitude, longitude, heading and altitude. Things show up as expected in the YAMCS web interface.

However, in the OpenMCT YAMCS plugin, the object tree displays the array as an individual telemetry point, and is therefore unable to expand to show the child elements.

Also, as the data is received, the clock updates, but it does not display in the table, it shows as unsupported data type.

Is there any support on this topic?

Thanks for your help.

**Comment by akhenry** [comment]
@akhenry

**Comment by akhenry** [comment]
@mk51tun We don't have great support for Arrays right now. Where possible parameter values of the Array type will be printed out as a string.

That doesn't help in your case though, where an array is part of the tree structure. Is it possible to use another named structure like a container, or a parent aggregate to wrap the child aggregates and make them available from the tree?

> Also, as the data is received, the clock updates, but it does not display in the table, it shows as unsupported data type.

Are you able to share a screen capture of what you're seeing? Did you start from the openmct-yamcs quickstart example? https://github.com/akhenry/openmct-yamcs/
